Project-Service-implementation-Script/main.py

```diff
@@ -0,0 +1,5238 @@
+    frameworkDocInsertObj['license']['orgDetails']['email'] = None
+    frameworkDocInsertObj['license']['orgDetails']['orgName'] = ccRootOrgName
+    frameworkDocInsertObj['license']['licenseDetails'] = {}
+    frameworkDocInsertObj['license']['licenseDetails']['name'] = "CC BY 4.0"
+    frameworkDocInsertObj['license']['licenseDetails']['url'] = "https://creativecommons.org/licenses/by/4.0/legalcode"
+    frameworkDocInsertObj['license']['licenseDetails']['description'] = "For details see below:"
+ 
+    urlCreateFrameworkApi = config.get(environment, 'INTERNAL_KONG_IP') + config.get(environment, 'frameworkCreationApi')
+    frameworkFilePath = solutionName_for_folder_path + '/framework/'
+    file_exists_framework = os.path.isfile(solutionName_for_folder_path + '/framework/uploadFile.json')
+    if not os.path.exists(frameworkFilePath):
+        os.mkdir(frameworkFilePath)
+
+    with open(frameworkFilePath + "uploadFile.json", "w",encoding='utf-8') as outfile:
+        json.dump(frameworkDocInsertObj, outfile)
+    headerFrameworkUploadApi = {'Authorization': config.get(environment, 'Authorization'),
+                                'X-authenticated-user-token': accessToken,
+                                'X-Channel-id': config.get(environment, 'X-Channel-id')}
+    filesFramework = {'framework': open(solutionName_for_folder_path + '/framework/uploadFile.json', 'rb')}
+
+    responseFrameworkUploadApi = requests.post(url=urlCreateFrameworkApi, headers=headerFrameworkUploadApi,
+                                               files=filesFramework)
+    messageArr = ["Framwork json file created.",
+                  "File loc : " + solutionName_for_folder_path + '/framework/uploadFile.json',
+                  "Framework upload API called,", "Status code : " + str(responseFrameworkUploadApi.status_code)]
+    createAPILog(solutionName_for_folder_path, messageArr)
+    if responseFrameworkUploadApi.status_code == 200:
+        print('Framework upload Success')
+        return frameworkExternalId
+
+    else:
+        messageArr = ["Framwork upload Failed.", "Response : " + responseFrameworkUploadApi.text]
+        createAPILog(solutionName_for_folder_path, messageArr)
+        print('Framework upload api failed in ' + environment,
+              'status_code response from api is ' + str(responseFrameworkUploadApi.status_code))
+        sys.exit()
+
+
+def themesUpload(solutionName_for_folder_path, wbObservation, millisAddObs, accessToken, frameworkExternalId,obsWORubWS):
+    global dictCritLookUp
+    with open(solutionName_for_folder_path + '/criteriaUpload/uploadInternalIdsSheet.csv', 'r',encoding='utf-8') as criteriaInternalFile:
+        criteriaInternalReader = csv.DictReader(criteriaInternalFile)
+        for crit in criteriaInternalReader:
+            dictCritLookUp[crit['Criteria External Id']] = crit['Criteria Internal Id']
+    if obsWORubWS:
+        print("Themes Observation without rubrics with scores")
```

Comment:
@MallanagoudaB why we are having observation related code in implementation script? why is it required?